src/rp2040.ts

```diff
@@ -350,31 +372,52 @@ export class RP2040 {
   }
 
   step() {
-    this.core.executeInstruction();
+    this.core0.executeInstruction();
+    this.core1.executeInstruction();
   }
 
   execute() {
+    this.core0.stopped = false;
+    this.core1.stopped = false;
+    setTimeout(() => this.executeInternal(), 0);
+  }
+
+  private executeInternal() {
     this.clock.resume();
     this.executeTimer = null;
-    this.stopped = false;
-    for (let i = 0; i < 100000 && !this.stopped && !this.core.waiting; i++) {
-      this.core.executeInstruction();
-    }
-    if (!this.stopped) {
-      this.executeTimer = setTimeout(() => this.execute(), 0);
+    let idle = false;
+    for (let i = 0; i < 1000 && !idle; i++) {
+      idle = true;
+      if (!this.core0.stopped && !this.core0.waiting) {
+        idle = false;
+        this.isCore0Running = true;
+        this.core0.executeInstruction();
```

Comment:
This eventually gets the cores out of sync. Imagine core0 executing a number of 5 cycle instructions and core 1 executing a number of 1 cycle instructions, let things run a while, then look at the `cycles` of each core (which will have diverged a lot).

You'll need something like https://github.com/c1570/rp2040js/blob/3a754941c169d7215d8d97112c01a9d1f9d4e2c9/demo/dual-mcu.ts#L102 here.

Reply:
Yes, the two core execution time may diverge if they run a while if `stopped` or `waiting` flag raised. Currently I do not have good solution for it. But what you did may not be helpful(if I understand correctly).

If one core is stopped or waiting, these state will not changed until the next `setTimeout` call since there is only 1 thread in our execution environment. Then your code to call `core.step` actually do nothing.

Reply:
Something like this is needed in `cortex-m0-core.ts`:
```
  executeInstruction() {
    if (this.waiting) {
      this.cycles++;
      return;
    }
    ...
```
...and something like this in `rp2040.ts` (this is my local version with #117 already merged):
```
  step() {
    let core0StartCycles = this.core0.cycles;
    this.isCore0Running = true;
    this.core0.executeInstruction();
    this.isCore0Running = false;
    while(this.core1.cycles < this.core0.cycles) {
      this.core1.executeInstruction();
    }
    for(let cycle = core0StartCycles; cycle < this.core0.cycles; cycle++) {
      this.pio[0].step();
      this.pio[1].step();
    }
  }
```